### ISSUE TITLE
Android SDK Platform-Tools

### ISSUE THREAD

**Issue by Weirdo1312** [microsoft/winget-pkgs#4082]
# Package Requested

Manifest:

Fill out as much of the manifest as you can:

```YAML
Id: string # publisher.package format
Publisher: Google # the name of the publisher
Name: Android SDK Platform-Tools # the name of the application
Version: 30.0.4 # version numbering format
License: developer.android.com/license # the open source license or copyright
InstallerType: exe (standalone, no installer) # enumeration of supported installer types (exe, msi, msix, inno, wix, nullsoft, appx)
Installers:
  - Arch: string # enumeration of supported architectures
    Url: https://dl.google.com/android/repository/platform-tools_r30.0.4-windows.zip # path to download installation file
    Sha256: string # SHA256 calculated from installer
# ManifestVersion: 0.1.0
```

**Comment by denelon** [comment]
@barfin is this a duplicate of #3680?

**Comment by barfin** [comment]
> @barfin is this a duplicate of #3680?

No, Android SDK platform tools are "adb" and "fastboot" command line tools for interacting with android phones with usb debugging enabled
like flashing some rom or recovery or rebooting into a specific img or and other good stuff
it's not for making android apps like android studio is

**Comment by ptanmay143** [comment]
> 
> # Package Requested
> 
> Manifest:
> 
> Fill out as much of the manifest as you can:
> 
> ```yaml
> Id: string # publisher.package format
> Publisher: Google # the name of the publisher
> Name: Android SDK Platform-Tools # the name of the application
> Version: 30.0.4 # version numbering format
> License: developer.android.com/license # the open source license or copyright
> InstallerType: exe (standalone, no installer) # enumeration of supported installer types (exe, msi, msix, inno, wix, nullsoft, appx)
> Installers:
>   - Arch: string # enumeration of supported architectures
>     Url: https://dl.google.com/android/repository/platform-tools_r30.0.4-windows.zip # path to download installation file
>     Sha256: string # SHA256 calculated from installer
> # ManifestVersion: 0.1.0
> ```

The installer type is not exe. It is actually a zip which needs to be extracted and added to the PATH environment variable. Till winget gets zip install support, I don't think we can get Android platform tools being packaged here.

**Comment by barfin** [comment]
@denelon please remove "Needs feedback" tag

**Comment by MattBDev** [comment]
This should have the label ".zip" and "Blocking-Issue"

**Comment by denelon** [comment]
https://github.com/microsoft/winget-cli/issues/859

**Comment by itsmeboy0** [comment]
> [microsoft/winget-cli#859](https://github.com/microsoft/winget-cli/issues/859)

Shouldn't it be https://github.com/microsoft/winget-cli/issues/140? (its not an installer inside)

**Comment by denelon** [comment]
We will have support for a portable app:

* microsoft/winget-cli/issues/182

first, then we will implement:

* microsoft/winget-cli/issues/140

which will include:

* microsoft/winget-cli/issues/859

**Comment by j2ghz** [comment]
> We will have support for a portable app:
> 
>    * [Support for installation of portable/standalone apps winget-cli#182](https://github.com/microsoft/winget-cli/issues/182)
> 
> 
> first, then we will implement:
> 
>    * [Support installing .zip files winget-cli#140](https://github.com/microsoft/winget-cli/issues/140)
> 
> 
> which will include:
> 
>    * [Support for installers inside of a .zip file winget-cli#859](https://github.com/microsoft/winget-cli/issues/859)

These seem to be implemented, is this unblocked now?

**Comment by vedantmgoyal2009** [comment]
cc @Trenly

**Comment by Trenly** [comment]
> > We will have support for a portable app:
> > 
> > * [Support for installation of portable/standalone apps winget-cli#182](https://github.com/microsoft/winget-cli/issues/182)
> > 
> > first, then we will implement:
> > 
> > * [Support installing .zip files winget-cli#140](https://github.com/microsoft/winget-cli/issues/140)
> > 
> > which will include:
> > 
> > * [Support for installers inside of a .zip file winget-cli#859](https://github.com/microsoft/winget-cli/issues/859)
> 
> These seem to be implemented, is this unblocked now?

Not exactly. They have been implemented, but not yet released. I can create the manifest for it, but it won’t be able to be merged until the Winget Client version 1.4 is released to stable and about 50% of users have updated to the new version

**Comment by bet4it** [comment]
Could it be implemented now?

**Comment by sitiom** [comment]
Nope not yet: https://github.com/microsoft/winget-pkgs/pull/94818#issuecomment-1406695813

**Comment by denelon** [comment]
* https://github.com/microsoft/winget-pkgs/issues/87300

**Comment by sitiom** [comment]
The problem is missing DLLs due to symlinking executables. Is there an open issue for this in winget-cli? If not, one should be made.

**Comment by Trenly** [comment]
Based on the discussion in the linked PR, this is a Zipped binary file and not just a portable EXE. Because of the way symlinks work, this is now blocked by https://github.com/microsoft/winget-cli/issues/2711

cc @denelon for labeling with Zipped Binary

**Comment by Trenly** [comment]
[Policy] Unblocked; Will require `ManifestVersion: 1.9.0` with `ArchiveBinariesDependOnPath: true`

**Comment by aisgbnok** [comment]
This has been resolved for all versions of the [`Google.PlatformTools`](https://github.com/microsoft/winget-pkgs/tree/master/manifests/g/Google/PlatformTools) package, which includes the latest release.

- #190804
